crates/build/src/buildinfo.rs

```diff
@@ -0,0 +1,288 @@
+    /// assert_eq!(Some("test"), BuildInfo::parse_uri("test").unwrap().name());
+    /// assert_eq!(Some("test"), BuildInfo::parse_uri("test?a=b").unwrap().name());
+    /// assert_eq!(None, BuildInfo::parse_uri("a=b").unwrap().name());
+    /// ```
+    pub fn name(&self) -> Option<&str> {
+        self.get("NAME")
+    }
+
+    /// Extract one of the [BuildInfo] values.
+    ///
+    /// ```
+    /// # use sightglass_build::BuildInfo;
+    /// assert_eq!(Some("test"), BuildInfo::parse_uri("test").unwrap().get("NAME"));
+    /// assert_eq!(Some("b"), BuildInfo::parse_uri("test?a=b").unwrap().get("a"));
+    /// assert_eq!(None, BuildInfo::parse_uri("a=b").unwrap().get("c"));
+    /// ```
+    pub fn get(&self, variable: &str) -> Option<&str> {
+        self.0.get(variable).map(String::as_str)
+    }
+
+    /// Modify one of the [BuildInfo] values.
+    ///
+    /// ```
+    /// # use sightglass_build::BuildInfo;
+    /// let mut buildinfo = BuildInfo::parse_uri("a").unwrap();
+    /// buildinfo.set("NAME", "b");
+    /// assert_eq!("b", buildinfo.as_uri());
+    /// ```
+    pub fn set<S: Into<String>>(&mut self, variable: S, value: S) -> Option<String> {
+        self.0.insert(variable.into(), value.into())
+    }
+
+    /// Modify one of the [BuildInfo] values.
+    ///
+    /// ```
+    /// # use sightglass_build::BuildInfo;
+    /// let buildinfo = BuildInfo::parse_uri("a").unwrap();
+    /// let mut iter = buildinfo.iter();
+    /// assert_eq!(Some(("NAME", "a")), iter.next());
+    /// assert_eq!(None, iter.next());
+    /// ```
+    pub fn iter<'a>(&'a self) -> impl Iterator<Item = (&'a str, &'a str)> {
+        self.0.iter().map(|(a, b)| (a.as_str(), b.as_str()))
+    }
+
+    /// Parse [BuildInfo] from a URI-like string; e.g., `<name>`,
+    /// `<name>?<var1>=<val1>+<var2>=<val2>`
+    ///
+    /// ```
+    /// # use sightglass_build::BuildInfo;
+    /// assert_eq!("test", BuildInfo::parse_uri("test").unwrap().as_uri());
+    /// assert_eq!("a=b", BuildInfo::parse_uri("a=b").unwrap().as_uri());
+    /// assert_eq!("test?a=b", BuildInfo::parse_uri("test?a=b").unwrap().as_uri());
+    /// assert_eq!("test?a=b+c=d", BuildInfo::parse_uri("test?a=b+c=d").unwrap().as_uri());
+    /// assert_eq!("test?a=b+c='d e'", BuildInfo::parse_uri("test?a=b+c='d e'").unwrap().as_uri());
+    /// ```
+    pub fn parse_uri(uri: &str) -> Result<Self> {
+        if let Some((name, rest)) = uri.split_once("?") {
+            let mut b = BuildInfo::from_iter(rest.split("+").map(|p| split_pair_str(p)));
+            b.0.insert("NAME".to_string(), name.to_string());
+            Ok(b)
+        } else if uri.contains("=") {
+            Ok(BuildInfo::from_iter(
+                uri.split("+").map(|p| split_pair_str(p)),
+            ))
+        } else {
+            Ok(BuildInfo::from_iter([("NAME", uri)]))
+        }
+    }
+
+    /// Emit [BuildInfo] as a URI-like string; e.g., `<name>`,
+    /// `<name>?<var1>=<val1>+<var2>=<val2>`. See [`Self::parse_uri()`].
```

Comment:
No, I ended up choosing to use `+` because `&` on the command line sends tasks to the background. It would have been nicer to use actual URI syntax but having to remember to quote all sightglass commands seemed like too much of an ask (plus all the troubleshooting when someone can't figure out why their command ran in the background with only the first build info parameter).